server/app/services/applicant/ApplicantService.java

```diff
@@ -551,14 +552,27 @@ private ApplicationPrograms relevantProgramsForApplicant(
               maybeSubmittedApp.map(Application::getSubmitTime);
           if (maybeDraftApp.isPresent()) {
             inProgressPrograms.add(
-                ApplicantProgramData.create(
-                    maybeDraftApp.get().getProgram().getProgramDefinition(),
-                    latestSubmittedApplicationTime));
+                ApplicantProgramData.builder()
+                    .setProgram(maybeDraftApp.get().getProgram().getProgramDefinition())
+                    .setLatestSubmittedApplicationTime(latestSubmittedApplicationTime)
+                    .build());
             programNamesWithApplications.add(programName);
           } else if (maybeSubmittedApp.isPresent() && activeProgramNames.containsKey(programName)) {
             ProgramDefinition programDef = activeProgramNames.get(programName);
+            Optional<StatusDefinitions.Status> maybeCurrentStatus =
+                programDef.statusDefinitions().getStatuses().stream()
+                    .filter(
+                        programStatus ->
+                            programStatus
+                                .statusText()
+                                .equals(maybeSubmittedApp.get().getLatestStatus().orElse("")))
```

Comment:
If getLatestStatus is empty then you can short circuit the above instead of doing the comparisons.